pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -1422,11 +1370,16 @@ func convertGateways(r ConfigContext) ([]config.Config, map[parentKey][]*parentI
 		for i, l := range kgw.Listeners {
 			i := i
 			namespaceLabelReferences.InsertAll(getNamespaceLabelReferences(l.AllowedRoutes)...)
-			server, ok := buildListener(r, obj, l, i)
+			server, ok := buildListener(r, obj, l, i, controllerName)
 			if !ok {
 				invalidListeners = append(invalidListeners, l.Name)
 				continue
 			}
+			servers = append(servers, server)
+			if controllerName == constants.ManagedGatewayMeshController {
+				// Waypoint doesn't actually convert the routes to VirtualServices
+				continue
```

Comment:
will waypoint ever support other fields like routes?  Wondering if istio needs to provide any sort of `unsupported fields` warnings for `istio-waypoint` gateways?

Reply:
It might, but not right now